### PR TITLE
Assignments/array hofs

### PR DESCRIPTION
**Pull request by EllisWaterman**

### CODE REVIEW

File: assignments/array-hofs/code.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+const grades = (array) => {
+  return array.map((n) => n.grade)
+}
+
+const pairs = (array) => {
+  return array.map((n) => [n,n])
+}
+
+const averageGrade = (array) => {
+  return (array.map((n) => n.grade)).reduce((total, n) => total+n)/array.length
```
Comment (gigamonkey) [inline assignments/array-hofs/code.js:26]: Just to point out, that the `.` is method calls has high left-to-right precedence so you can chain method calls without `()`s. E.g.

```javascript
  return array.map((n) => n.grade).reduce((total, n) => total+n) / array.length
```

File: assignments/array-hofs/code.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+const allEven = (array) => {
+  return array.every((n) => n % 2 === 0)
+}
+
+const someEven = (array) => {
+  return array.some((n) => n % 2 === 0)
+}
+
+const lengthOfNames = (array) => {
+  return (array.filter((n) => n[0].toUpperCase() === n[0])).map((n) => n.length)
```
Comment (gigamonkey) [inline assignments/array-hofs/code.js:42]: Again, don't need the extra `()`s when chaining method calls.

**Review by gigamonkey — APPROVED** [review]
Looks good. One style comment below.

File: assignments/array-hofs/code.js (hunk 2 above)
Comment (EllisWaterman) [reply assignments/array-hofs/code.js]: done

**Pull request merged**